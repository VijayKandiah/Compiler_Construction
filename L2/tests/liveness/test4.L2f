(:myF
  2 0

  %myVar1 <- 5
  %myVar2 <- 0

  cjump rdi < rsi :trueLabel :falseLabel
:trueLabel
  rax <- %myVar1
  goto :endLabel
:falseLabel
  rax <- %myVar2

:endLabel
  return
)
