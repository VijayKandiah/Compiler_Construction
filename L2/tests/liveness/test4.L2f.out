(
(in
(r12 r13 r14 r15 rbp rbx rdi rsi)
(myVar1 r12 r13 r14 r15 rbp rbx rdi rsi)
(myVar1 myVar2 r12 r13 r14 r15 rbp rbx rdi rsi)
(myVar1 r12 r13 r14 r15 rbp rbx)
(myVar1 r12 r13 r14 r15 rbp rbx)
(r12 r13 r14 r15 rax rbp rbx)
(myVar2 r12 r13 r14 r15 rbp rbx)
(myVar2 r12 r13 r14 r15 rbp rbx)
(r12 r13 r14 r15 rax rbp rbx)
(r12 r13 r14 r15 rax rbp rbx)
)

(out
(myVar1 r12 r13 r14 r15 rbp rbx rdi rsi)
(myVar1 myVar2 r12 r13 r14 r15 rbp rbx rdi rsi)
(myVar1 myVar2 r12 r13 r14 r15 rbp rbx)
(myVar1 r12 r13 r14 r15 rbp rbx)
(r12 r13 r14 r15 rax rbp rbx)
(r12 r13 r14 r15 rax rbp rbx)
(myVar2 r12 r13 r14 r15 rbp rbx)
(r12 r13 r14 r15 rax rbp rbx)
(r12 r13 r14 r15 rax rbp rbx)
()
)

)
